File: uniffi/src/message/message_list.rs
use cesrox_core::error::CesrResult;

use cesrox_core::{
    MessageList as CesroxMessageList,
};
use crate::message::message::Message;

/*
    MessageList
*/
pub struct MessageList {
    pub messages: Vec<Message>,
}

pub struct MessageListFromStreamResult {
    pub rest: Vec<u8>,
    pub messages: Vec<Message>
}

pub fn message_list_from_stream_bytes(bytes: &[u8]) -> CesrResult<MessageListFromStreamResult> {
    let (rest, message_list) = CesroxMessageList::from_stream_bytes(bytes)?;
    let messages = message_list.messages.into_iter().map(|message| Message::from(message)).collect();
    Ok(MessageListFromStreamResult {
        rest: rest.to_vec(),
        messages
    })
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/MessageList.kt
package cesrox

import uniffi.cesrox.*

class MessageList(val rest: List<UByte>, val messages: List<Message>) {
    companion object {
        fun fromBytes(bytes: List<UByte>): MessageList {
            val result: MessageListFromStreamResult = messageListFromStreamBytes(bytes)
            return MessageList(result.rest, result.messages.map{ Message(it) })
        }
    }
}